lte/gateway/c/core/oai/tasks/amf/amf_app_state_converter.cpp

```diff
@@ -436,5 +499,323 @@ void AmfNasStateConverter::proto_to_amf_security_context(
   state_amf_security_context->direction_decode =
       emm_security_context_proto.direction_decode();
 }
+void AmfNasStateConverter::smf_proc_data_to_proto(
+    const smf_proc_data_t* state_smf_proc_data,
+    magma::lte::oai::Smf_Proc_Data* smf_proc_data_proto) {
+  smf_proc_data_proto->set_pdu_session_id(state_smf_proc_data->pdu_session_id);
+  smf_proc_data_proto->set_pti(state_smf_proc_data->pti);
+  smf_proc_data_proto->set_message_type(
+      static_cast<uint32_t>(state_smf_proc_data->message_type));
+  smf_proc_data_proto->set_max_uplink(state_smf_proc_data->max_uplink);
+  smf_proc_data_proto->set_max_downlink(state_smf_proc_data->max_downlink);
+  smf_proc_data_proto->set_pdu_session_type(
+      (magma::lte::oai::M5GPduSessionType)
+          state_smf_proc_data->pdu_session_type);
+  smf_proc_data_proto->set_ssc_mode(state_smf_proc_data->ssc_mode);
+}
+void AmfNasStateConverter::proto_to_smf_proc_data(
+    const magma::lte::oai::Smf_Proc_Data& smf_proc_data_proto,
+    smf_proc_data_t* state_smf_proc_data) {
+  state_smf_proc_data->pdu_session_id = smf_proc_data_proto.pdu_session_id();
+  state_smf_proc_data->pti            = smf_proc_data_proto.pti();
+  state_smf_proc_data->message_type =
+      static_cast<M5GMessageType>(smf_proc_data_proto.message_type());
+  state_smf_proc_data->max_uplink   = smf_proc_data_proto.max_uplink();
+  state_smf_proc_data->max_downlink = smf_proc_data_proto.max_downlink();
+  state_smf_proc_data->pdu_session_type =
+      static_cast<M5GPduSessionType>(smf_proc_data_proto.pdu_session_type());
+  state_smf_proc_data->ssc_mode = smf_proc_data_proto.ssc_mode();
+}
+
+void AmfNasStateConverter::s_nssai_to_proto(
+    const s_nssai_t* state_s_nssai, magma::lte::oai::SNssai* snassi_proto) {
+  snassi_proto->set_sst(state_s_nssai->sst);
+  snassi_proto->set_sd(*(uint32_t*) &state_s_nssai->sd);
+}
+void AmfNasStateConverter::proto_to_s_nssai(
+    const magma::lte::oai::SNssai& snassi_proto, s_nssai_t* state_s_nssai) {
+  state_s_nssai->sst              = snassi_proto.sst();
+  *(uint32_t*) &state_s_nssai->sd = snassi_proto.sd();
```

Comment:
Rebase on #10958 , which changes this code

Reply:
Comment Addressed